src/monio/AtlasWriter.h

```diff
@@ -35,64 +37,76 @@ class AtlasWriter {
   AtlasWriter& operator=( AtlasWriter&&)      = delete;  //!< Deleted move assignment
   AtlasWriter& operator=(const AtlasWriter&)  = delete;  //!< Deleted copy assignment
 
-  // For writing of FieldSets with no metadata
-  void populateFileDataWithField(FileData& fileData,
-                           const atlas::Field& field,
-                           const std::string& writeName);
-
-  // For writing LFRic data with some existing metadata
+  /// \brief Creates required metadata and data from at Atlas field. For writing LFRic data with
+  ///        some existing metadata.
   void populateFileDataWithField(FileData& fileData,
                                  atlas::Field& field,
                            const consts::FieldMetadata& fieldMetadata,
                            const std::string& writeName,
                            const bool isLfricNaming);
 
- private:
-  void populateDataContainerWithField(std::shared_ptr<monio::DataContainerBase>& dataContainer,
-                                const atlas::Field& field,
-                                const std::vector<size_t>& lfricToAtlasMap,
-                                const std::string& fieldName);
-
-  void populateDataContainerWithField(std::shared_ptr<monio::DataContainerBase>& dataContainer,
-                                const atlas::Field& field,
-                                const std::vector<int>& dimensions);
-
+  /// \brief Creates all metadata and data from at Atlas field. For writing of field sets with no
+  ///        metadata.
+  void populateFileDataWithField(FileData& fileData,
+                           const atlas::Field& field,
+                           const std::string& writeName);
 
+ private:
+  /// \brief Called from the entry point with LFRic metadata.
   void populateMetadataWithField(Metadata& metadata,
                            const atlas::Field& field,
                            const consts::FieldMetadata& fieldMetadata,
                            const std::string& varName);
 
+  /// \brief Called from the entry point with no existing metadata.
   void populateMetadataWithField(Metadata& metadata,
                            const atlas::Field& field,
                            const std::string& varName);
 
+  /// \brief Called from the entry point with LFRic metadata.
   void populateDataWithField(Data& data,
                        const atlas::Field& field,
                        const std::vector<size_t>& lfricToAtlasMap,
                        const std::string& fieldName);
 
+  /// \brief Called from the entry point with no existing metadata.
   void populateDataWithField(Data& data,
                        const atlas::Field& field,
                        const std::vector<int> dimensions);
 
+  /// \brief With LFRic metadata, derives the container type and makes the call to populate it.
```

Comment:
Slightly confusing since the other functions use metadata in reference to `Metadata` or `consts::FieldMetadata`. I find it confusing at least :D
The mapping is a kind of metadata, but I would say it's different to variable names, etc. Up to you
```suggestion
  /// \brief With LFRic mapping, derives the container type and makes the call to populate it.
```

Reply:
Thanks Josh. Yeah, I can see the source of the confusion. Hopefully my updated comments will get across what I was aiming for with more clarity. See what you think.